plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package rbac implements the authorizer.Authorizer interface using roles base access control.
+package rbac
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/pkg/apis/rbac/validation"
+	"k8s.io/kubernetes/pkg/auth/authorizer"
+	"k8s.io/kubernetes/pkg/auth/user"
+)
+
+type RBACAuthorizer struct {
+	context                   api.Context
+	authorizationRuleResolver validation.AuthorizationRuleResolver
+}
+
+func (r *RBACAuthorizer) Authorize(attr authorizer.Attributes) error {
+	userInfo := &user.DefaultInfo{
+		Name:   attr.GetUserName(),
+		Groups: attr.GetGroups(),
+	}
+
+	ctx := api.WithNamespace(api.WithUser(r.context, userInfo), attr.GetNamespace())
+
+	// Frame the authorization request as a privilege escalation check.
+	var requestedRule *rbac.PolicyRule
+	if attr.IsResourceRequest() {
```

Comment:
This looks backwards to me? Shouldn't a rule with `Verbs`, `Resources`, etc. be constructed for resource requests?